pkg/operator/target_config_reconciler.go

```diff
@@ -0,0 +1,274 @@
+)
+
+type TargetConfigReconciler struct {
+	targetImagePullSpec string
+	operatorNamespace   string
+
+	jobSetOperatorClient       *operatorclient.JobSetOperatorClient
+	kubeClient                 kubernetes.Interface
+	apiextensionsClient        *apiextensionsclient.Clientset
+	eventRecorder              events.Recorder
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces
+	jobSetOperatorsLister      openshiftoperatorv1.JobSetOperatorLister
+	secretsLister              v1.SecretLister
+	resourceCache              resourceapply.ResourceCache
+}
+
+func NewTargetConfigReconciler(targetImagePullSpec, operatorNamespace string,
+	operatorClientInformer operatorclientinformers.JobSetOperatorInformer,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	jobSetOperatorClient *operatorclient.JobSetOperatorClient,
+	kubeClient kubernetes.Interface,
+	apiextensionsClient *apiextensionsclient.Clientset,
+	eventRecorder events.Recorder,
+) factory.Controller {
+	t := &TargetConfigReconciler{
+		targetImagePullSpec:   targetImagePullSpec,
+		operatorNamespace:     operatorNamespace,
+		jobSetOperatorClient:  jobSetOperatorClient,
+		kubeClient:            kubeClient,
+		apiextensionsClient:   apiextensionsClient,
+		secretsLister:         kubeInformersForNamespaces.SecretLister(),
+		jobSetOperatorsLister: operatorClientInformer.Lister(),
+		eventRecorder:         eventRecorder,
+		resourceCache:         resourceapply.NewResourceCache(),
+	}
+	return factory.New().WithInformers(
+		// for the operator changes
+		operatorClientInformer.Informer(),
+		// for the deployment and its configmap and secret
+		kubeInformersForNamespaces.InformersFor(operatorNamespace).Apps().V1().Deployments().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorNamespace).Core().V1().ConfigMaps().Informer(),
+		kubeInformersForNamespaces.InformersFor(operatorNamespace).Core().V1().Secrets().Informer(),
+	).ResyncEvery(time.Minute*5).WithSync(t.sync).ToController("TargetConfigController", eventRecorder)
+}
+
+func (t TargetConfigReconciler) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	jobSetOperator, err := t.jobSetOperatorsLister.JobSetOperators(t.operatorNamespace).Get(operatorclient.OperatorConfigName)
+	if err != nil {
+		klog.ErrorS(err, "unable to get operator configuration", "namespace", t.operatorNamespace, "jobset", operatorclient.OperatorConfigName)
+		return err
+	}
+
+	ownerReference := metav1.OwnerReference{
+		APIVersion: "operator.openshift.io/v1",
+		Kind:       "JobSetOperator",
+		Name:       jobSetOperator.Name,
+		UID:        jobSetOperator.UID,
+	}
+
+	specAnnotations := map[string]string{
+		"jobsetoperators.operator.openshift.io/cluster": strconv.FormatInt(jobSetOperator.Generation, 10),
+	}
+
+	_, _, err = t.manageWebhookService(ctx, ownerReference)
+	if err != nil {
+		return err
+	}
+	webhookSecret, _, err := t.manageWebhookCertSecret()
```

Comment:
I'm wondering why there is this secret and not in LWS. I assume this comes from upstream of JobSet?